libcxx/docs/TestingLibcxx.rst

```diff
@@ -180,6 +180,38 @@ The tests of libc++ are stored in libc++'s testing related subdirectories:
   ``libcxx/test/libcxx``. The structure of this directories follows the
   structure of ``libcxx/test/std``.
 
+Principles of testing
+---------------------
+
+Tests are a practical way to validate the correctness of the code. As such, they contain pragmatic trade offs between
+the cost of writing and maintaining the tests and the value they provide. Please consider the following principles when
+writing tests:
+
+- **Consider the next reader**
+
+    Tests should be obvious to the future reader. Avoid too much boiler plate or other
```

Comment:
We should avoid terms like "obvious", as that's a subjective metric. Things that are obvious to you might not be obvious to me, and vice versa.
```suggestion
    Unit tests should be written so that anyone who lacks context can immediately identify that they are correct. Avoid too much boiler plate or other
```